Makefile

```diff
@@ -361,11 +337,15 @@ ifdef main
 	make publish_langflow
 endif
 
+check_tools: ## check for required tools
+	@command -v poetry >/dev/null 2>&1 || { echo >&2 "Poetry is not installed. Aborting."; exit 1; }
+	@command -v npm >/dev/null 2>&1 || { echo >&2 "NPM is not installed. Aborting."; exit 1; }
+	@command -v docker >/dev/null 2>&1 || { echo >&2 "Docker is not installed. Aborting."; exit 1; }
+	@command -v pipx >/dev/null 2>&1 || { echo >&2 "pipx is not installed. Aborting."; exit 1; }
```

Comment:
I like this command! However, I think there should be a success message at the end after all checks pass. Can you add that?